File: sentinelai-frontend/src/types/notification.ts
export interface Notification {
  id: string
  event_type: string
  title: string
  message?: string
  severity: 'critical' | 'warning' | 'info' | 'success'
  source?: string
  source_id?: string
  is_read: boolean
  created_at: string
}

export interface NotificationListResponse {
  items: Notification[]
  total: number
}

export interface UnreadCountResponse {
  count: number
}

export interface NotificationPreferences {
  email_enabled: boolean
  desktop_enabled: boolean
  slack_enabled: boolean
  discord_enabled: boolean
  teams_enabled: boolean
  telegram_enabled: boolean
  critical_only: boolean
  muted_until?: string | null
  event_subscriptions: Record<string, boolean>
}

export type WebSocketStatus = 'connecting' | 'connected' | 'disconnected' | 'error'

export interface LiveActivityEvent {
  type: 'notification' | 'event'
  event: string
  notification?: Notification
  data?: Record<string, unknown>
  timestamp?: string
}

export interface SystemStatus {
  websocket: WebSocketStatus
  backend: 'healthy' | 'degraded' | 'down'
  database: 'healthy' | 'degraded' | 'down'
  redis: 'healthy' | 'degraded' | 'down'
}

File: sentinelai-backend/app/models/notification.py
from datetime import datetime
from sqlalchemy import String, Text, DateTime, Boolean, Integer
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import Mapped, mapped_column

from app.database.base import BaseModel


class Notification(BaseModel):
    __tablename__ = "notifications"

    user_id: Mapped[str] = mapped_column(String(255), nullable=False, index=True)
    event_type: Mapped[str] = mapped_column(String(100), nullable=False, index=True)
    title: Mapped[str] = mapped_column(String(500), nullable=False)
    message: Mapped[str | None] = mapped_column(Text, nullable=True)
    severity: Mapped[str] = mapped_column(String(20), default="info", nullable=False)
    source: Mapped[str | None] = mapped_column(String(100), nullable=True)
    source_id: Mapped[str | None] = mapped_column(String(255), nullable=True)
    is_read: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    read_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    metadata_json: Mapped[dict | None] = mapped_column(JSONB, nullable=True)


class NotificationPreference(BaseModel):
    __tablename__ = "notification_preferences"

    user_id: Mapped[str] = mapped_column(String(255), nullable=False, index=True, unique=True)
    email_enabled: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    desktop_enabled: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    slack_enabled: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    discord_enabled: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    teams_enabled: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    telegram_enabled: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    critical_only: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    muted_until: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    event_subscriptions: Mapped[dict | None] = mapped_column(JSONB, nullable=True)
